h2o-core/src/main/java/hex/ModelBuilder.java

```diff
@@ -1329,7 +1362,10 @@ else if (_parms._weights_column != null && _weights != null && !_weights.isBinar
         separateFeatureVecs(); //fix up the pointers to the special vecs
       }
       if (_valid != null) {
-        _valid = encodeFrameCategoricals(_valid, ! _parms._is_cv_model /* for CV, need to score one more time in outer loop */);
+        Frame encodedVa = FrameUtils.applyTargetEncoder(teModel, va);
+        _toDelete.put(encodedVa._key, Arrays.toString(Thread.currentThread().getStackTrace()));
```

Comment:
logic similar as for training frame + you also need to ensure that `teModel != null` here, so maybe it's worth having a private method similar to `encodeFrameCategoricals`.

Reply:
@sebhrusen yes, I was going to introduce one after discussion about which approach to use `enum` vs `separate`.  This would automatically solve missing `teModel == null` check in case I'm lucky enough to copy from right place :) Thanks!

Reply:
@sebhrusen fixed and covered with the tests. will introduce private method once other things are settled